src/test/java/uk/gov/companieshouse/documentstore/consumer/kafka/ConsumerInvalidPayloadExceptionIT.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.companieshouse.documentstore.consumer.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.companieshouse.documentstore.consumer.kafka.KafkaUtils.ERROR_TOPIC;
+import static uk.gov.companieshouse.documentstore.consumer.kafka.KafkaUtils.INVALID_TOPIC;
+import static uk.gov.companieshouse.documentstore.consumer.kafka.KafkaUtils.MAIN_TOPIC;
+import static uk.gov.companieshouse.documentstore.consumer.kafka.KafkaUtils.RETRY_TOPIC;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.ByteArrayOutputStream;
+import java.time.Duration;
+import org.apache.avro.io.DatumWriter;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.reflect.ReflectDatumWriter;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import uk.gov.companieshouse.documentstore.consumer.service.DeltaService;
+
+@SpringBootTest
+class ConsumerInvalidPayloadExceptionIT extends AbstractKafkaIT {
+
+    @Autowired
+    private KafkaConsumer<String, byte[]> testConsumer;
+    @Autowired
+    private KafkaProducer<String, byte[]> testProducer;
+    @Autowired
```

Comment:
Ln 37 to 43 aren't required for the test to pass. Can probably remove.

Reply:
resolved in commit